src/Features/CSharp/Portable/ConvertAnonymousTypeToClass/CSharpConvertAnonymousTypeToClassCodeRefactoringProvider.cs

```diff
@@ -41,7 +42,21 @@ private ArgumentListSyntax CreateArgumentList(AnonymousObjectCreationExpressionS
                 SyntaxFactory.Token(SyntaxKind.CloseParenToken).WithTriviaFrom(anonymousObject.CloseBraceToken));
 
         private SeparatedSyntaxList<ArgumentSyntax> CreateArguments(SeparatedSyntaxList<AnonymousObjectMemberDeclaratorSyntax> initializers)
-            => SyntaxFactory.SeparatedList<ArgumentSyntax>(CreateArguments(initializers.GetWithSeparators()));
+            => SyntaxFactory.SeparatedList<ArgumentSyntax>(CreateArguments(OmitTrailingComma(initializers.GetWithSeparators())));
+
+        private static SyntaxNodeOrTokenList OmitTrailingComma(SyntaxNodeOrTokenList list)
+        {
+            // Trailing comma is allowed in initializer list, but disallowed in method calls.
+            return list.Count >= 2 && list[^1].IsToken
+                ? list
+                    .Replace(
+                        list[^2],
+                        list[^2].AsNode()
+                            .WithAppendedTrailingTrivia(list[^1].GetLeadingTrivia())
+                            .WithAppendedTrailingTrivia(list[^1].GetTrailingTrivia()))
+                    .RemoveAt(list.Count - 1)
+                : list;
```

Comment:
this has gotten to bee too much of a mouthful.  Can you just have this be:
```suggestion
            if (list.Count % 2 == 1)
                return list;
                
            // now just return the list without the comma.
```

Reply:
`list.Count % 2 == 1` would break the empty list case, so we really need the `list.Count >= 2` condition. But I agree that it's better to get rid of the ternary conditional.

Reply:
sounds good.

Reply:
Fixed.